### PR TITLE
feat: [CHAOS-925]: Documentation updates for Chaos Module

### PR DESCRIPTION
**Pull request by SahilKr24**
Signed-off-by: Sahil <sahil@harness.io>

# Harness Developer Pull Request
Thanks for helping us make the Developer Hub better. The PR will be looked at
by the maintainers. 

## What Type of PR is This?

- [ ] Issue
- [ ] Feature
- [ ] Maintenance/Chore

If tied to an Issue, list the Issue(s) here:

* *Issue(s)*

## House Keeping
Some items to keep track of. Screen shots of changes are optional but would help the maintainers review quicker. 

- [ ] Tested Locally
- [ ] *Optional* Screen Shoot.

### CODE REVIEW

File: docs/chaos-engineering/chaos-faults/azure/azure-instance-memory-hog.md [hunk 1]
```diff
@@ -1,41 +1,32 @@
...
-### Uses of the experiment
-
-:::info
-
-- The fault causes memory hog/stress on the target Azure Instance(s). The idea of this fault is to simulate issues when there is lack of memory for other runnning processes/applications resulting into degrading their performance.
-- Injecting a rogue process into a target Azure instance, we starve the main processes/applications (typically pid 1) of the resources allocated to it (where limits are defined) causing slowness in application traffic or in other cases unrestrained use can cause instance to exhaust resources leading to degradation in performance of processes/applications present on the instance. So this category of chaos fault helps to build the immunity on the application undergoing any such stress scenario.
+<details>
+<summary>View fault usage</summary>
+<div>
+This fault determines the resilience of an Azure instance when memory resources are utilized in excess, unexpectedly. It determines how Azure scales the memory to maintain the application when resources are consumed heavily.
```
Comment (neelanjan00) [inline docs/chaos-engineering/chaos-faults/azure/azure-instance-memory-hog.md:19]: Do you want to mention the OOM kill scenarios?

File: docs/chaos-engineering/chaos-faults/azure/azure-instance-stop.md [hunk 2]
```diff
@@ -1,32 +1,27 @@
...
 
-## Uses
+## Usage
 
 <details>
-<summary>View the uses of the fault</summary>
+<summary>View fault usage</summary>
 <div>
-Coming soon.
+This fault determines the resilience of an application to unexpected power offs from Azure instances. It determines how the application handles the requests and how quickly it recovers from such failures.
```
Comment (neelanjan00) [inline docs/chaos-engineering/chaos-faults/azure/azure-instance-stop.md:15]:
```suggestion
This fault determines the resilience of an application to unexpected power-off of the Azure instances. It determines how the application handles the requests and how quickly it recovers from such failures.
```

File: docs/chaos-engineering/chaos-faults/azure/azure-web-app-stop.md [hunk 3]
```diff
@@ -1,32 +1,27 @@
...
-## Uses
+## Usage
 <details>
-<summary>View the uses of the fault</summary>
+<summary>View fault usage</summary>
 <div>
-Web App stop is another very frequent scenario we find with Azure web app service which stops a running web app and impacts its delivery. Such scenarios can still occur despite whatever availability aids web app service provides.
-
-Stopping the web app will disrupt its performance and impact the smooth working of the application. So this category of chaos fault helps to build immunity to the application undergoing any such scenarios.
+This fault determines the resilience of a web application to unplanned halts (or stops). It determines the resilience based on how quicly (and efficiently) the application recovers from the failure by re-routing the traffic to a different instance on the same application service.
```
Comment (neelanjan00) [inline docs/chaos-engineering/chaos-faults/azure/azure-web-app-stop.md:15]:
```suggestion
This fault determines the resilience of a web application to unplanned halts (or stops). It determines the resilience based on how quickly (and efficiently) the application recovers from the failure by re-routing the traffic to a different instance on the same application service. 
```

File: docs/chaos-engineering/chaos-faults/kubernetes/node/node-cpu-hog.md [hunk 4]
```diff
@@ -1,38 +1,35 @@
...
-:::
 
-## Uses
+
+## Usage
 <details>
 <summary>View the uses of the fault</summary>
 <div>
-Coming soon.
+The fault aims to verify resiliency of applications whose replicas may be evicted on account on nodes turning unschedulable (Not Ready) due to lack of CPU resources.
```
Comment (neelanjan00) [inline docs/chaos-engineering/chaos-faults/kubernetes/node/node-cpu-hog.md:17]:
```suggestion
The fault aims to verify the resiliency of applications whose replicas may be evicted on account of nodes turning unschedulable (Not Ready) or new replicas not being able to schedule due to a lack of CPU resources.
```

File: docs/chaos-engineering/chaos-faults/kubernetes/node/node-drain.md [hunk 5]
```diff
@@ -1,58 +1,55 @@
 ---
 id: node-drain
-title: Node Drain
+title: Node drain
 ---
-## Introduction
 
-- It drain the node. The resources which are running on the target node should be reschedule on the other nodes.
+Node drain drains the node of all its resources running on it.
```
Comment (neelanjan00) [inline docs/chaos-engineering/chaos-faults/kubernetes/node/node-drain.md:6]:
```suggestion
Node drain fault drains all the resources running on a node. 
```

File: docs/chaos-engineering/chaos-faults/kubernetes/node/node-drain.md [hunk 6]
```diff
@@ -1,58 +1,55 @@
...
-:::
 
-## Uses
+
+## Usage
 <details>
 <summary>View the uses of the fault</summary>
 <div>
-Coming soon.
+This fault determines the resilience of the application when the nodes are deprived of resources.
```
Comment (neelanjan00) [inline docs/chaos-engineering/chaos-faults/kubernetes/node/node-drain.md:16]:
```suggestion
This fault determines the resilience of the application when the application replicas scheduled on a node are removed. It validates the application failover capabilities when a node suddenly becomes unavailable.
```
Node Drain has nothing to do with the deprivation of resources.

File: docs/chaos-engineering/chaos-faults/kubernetes/node/node-io-stress.md [hunk 7]
```diff
@@ -1,39 +1,38 @@
...
-:::
 
-## Uses
+
+## Usage
 <details>
 <summary>View the uses of the fault</summary>
 <div>
-Coming soon.
+The fault aims to verify the resilience of applications that share the disk resource for ephemeral or persistent storage purposes.
```
Comment (neelanjan00) [inline docs/chaos-engineering/chaos-faults/kubernetes/node/node-io-stress.md:19]:
```suggestion
The fault aims to verify the resilience of applications that share the disk resource for ephemeral or persistent storage purposes during high disk I/O usage.
```

File: docs/chaos-engineering/chaos-faults/kubernetes/node/node-memory-hog.md [hunk 8]
```diff
@@ -1,35 +1,32 @@
 ---
 id: node-memory-hog
-title: Node Memory Hog
+title: Node memory hog
 ---
 
-## Introduction
+Node memory hog causes memory resource exhaustion on the Kubernetes node. 
+- It is injected using a helper pod running the Linux stress-ng tool (a workload generator).
+- The chaos affects the application foe a duration specified by the `TOTAL_CHAOS_DURATION` environment variable.
```
Comment (neelanjan00) [inline docs/chaos-engineering/chaos-faults/kubernetes/node/node-memory-hog.md:8]:
```suggestion
- The chaos affects the application for a duration specified by the `TOTAL_CHAOS_DURATION` environment variable.
```

File: docs/chaos-engineering/chaos-faults/kubernetes/node/node-memory-hog.md [hunk 9]
```diff
@@ -1,35 +1,32 @@
...
 
-## Uses
+
+## Usage
 <details>
-<summary>View the uses of the fault</summary>
+<summary>View fault usage</summary>
 <div>
-Coming soon.
+Node memory hog causes memory resource exhaustion on the Kubernetes node. The fault aims to verify resilience of applications whose replicas may be evicted on account on nodes becoming unschedulable (Not Ready) due to lack of memory resources.
```
Comment (neelanjan00) [inline docs/chaos-engineering/chaos-faults/kubernetes/node/node-memory-hog.md:17]: Do you want to discuss about the OOMKill scenarios here?

File: docs/chaos-engineering/chaos-faults/kubernetes/node/node-restart.md [hunk 10]
```diff
@@ -1,64 +1,62 @@
...
-:::
 
-## Uses
+
+## Usage
 <details>
 <summary>View the uses of the fault</summary>
 <div>
-Coming soon.
+This fault determines the deployment sanity (replica availability and uninterrupted service) and recovery workflows of the application pod.
```
Comment (neelanjan00) [inline docs/chaos-engineering/chaos-faults/kubernetes/node/node-restart.md:17]:
```suggestion
This fault determines the deployment sanity (replica availability and uninterrupted service) and recovery workflows of the application pod in the event of an unexpected node restart.
```

File: docs/chaos-engineering/chaos-faults/kubernetes/node/node-taint.md [hunk 11]
```diff
@@ -1,40 +1,37 @@
 ---
 id: node-taint
-title: Node Taint
+title: Node taint
 ---
 
-## Introduction
-- It taints the node to apply the desired effect. The resources which contains the correspoing tolerations can only bypass the taints.
+Node taint taints (contaminates) the node by applying the desired effect.
```
Comment (neelanjan00) [inline docs/chaos-engineering/chaos-faults/kubernetes/node/node-taint.md:6]: Why are we talking about contamination here?

File: docs/chaos-engineering/chaos-faults/kubernetes/node/node-taint.md [hunk 12]
```diff
@@ -1,40 +1,37 @@
...
 ![Node Taint](./static/images/node-taint.png)
-:::
 
-## Uses
+## Usage
 <details>
 <summary>View the uses of the fault</summary>
 <div>
-Coming soon.
+The fault aims to verify resiliency of applications when the nodes of a container (or deployment) are contaminated.
```
Comment (neelanjan00) [inline docs/chaos-engineering/chaos-faults/kubernetes/node/node-taint.md:15]:
```suggestion
The fault aims to verify the resiliency of applications when a certain taint is added to a node.
```

File: docs/chaos-engineering/chaos-faults/kubernetes/pod/container-kill.md [hunk 13]
```diff
@@ -1,39 +1,36 @@
...
 
-## Uses
+
+## Usage
 <details>
-<summary>View the uses of the fault</summary>
+<summary>View fault usage</summary>
 <div>
-Coming soon.
+It tests an application's deployment sanity (replica availability and uninterrupted service) and recovery workflow.
```
Comment (neelanjan00) [inline docs/chaos-engineering/chaos-faults/kubernetes/pod/container-kill.md:17]:
```suggestion
It tests an application's deployment sanity (replica availability and uninterrupted service) and recovery workflow when certain replicas are not available.
```

**Review by neelanjan00 — APPROVED** [review]

**Comment by SmritiSatya** [comment]
Sure, Kim. Thank you.

**Pull request merged**